### PR TITLE
[IMP] test_lint: check for function-redefined error with pylint

### PR DESCRIPTION
**Pull request by Elkasitu**
With this commit, pylint will now point out when the same
function/class/method is defined more than once in the same scope which
is a recurring mistake

Linked to https://github.com/odoo/enterprise/pull/14090

Prevents errors such as https://github.com/odoo/odoo/issues/59683 (and many others) from even happening

### CODE REVIEW

File: addons/account_payment/controllers/portal.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo.addons.account.controllers.portal import PortalAccount
+from odoo.addons.account.controllers.portal import PortalAccount as PA
 from odoo.http import request
 
 
-class PortalAccount(PortalAccount):
+class PortalAccount(PA):
```
Comment (tde-banana-odoo) [inline addons/account_payment/controllers/portal.py:8]: Why not real names likes AccountPortalAccount (well this one sounds weird, but you got the idea :D)

Comment (Elkasitu) [reply addons/account_payment/controllers/portal.py:8]: I tried that approach originally but then realized something: I'd have to also change all imports and inheritances in downstream modules and potentially break third-party apps too :scream:

E.g., if module `account_foo` imports `account_payment`'s `PortalAccount`, I'll have to rename it to `AccountPortalAccount` in `account_payment` and in `account_foo` too

I can change it depending on what the general consensus is, I just did it the lazy way for now :stuck_out_tongue_winking_eye:

Comment (tde-banana-odoo) [reply addons/account_payment/controllers/portal.py:8]: Indeed. Minimal patch.

And the reverse move ? Import PortalAccount as BasePortalAccount ?

File: odoo/addons/test_new_api/tests/test_new_fields.py [hunk 2]
```diff
@@ -573,16 +573,6 @@ def test_13_inverse_access(self):
         with self.assertRaises(AccessError):
             foo.with_user(user).display_name = 'Forbidden'
 
-    def test_13_inverse_access(self):
```
Comment (JKE-be) [inline odoo/addons/test_new_api/tests/test_new_fields.py:576]: seems not related to the current PR ;)

Comment (Elkasitu) [reply odoo/addons/test_new_api/tests/test_new_fields.py:576]: it is, this patch detects function/method/etc redefinitions, this test is defined twice, so only the second one sequentially is every executed (in this case both are identical so I just removed one of them), probably a forward-port bug

File: addons/account_payment/controllers/portal.py (hunk 1 above)
Comment (Elkasitu) [reply addons/account_payment/controllers/portal.py:8]: Also works, I did that for some of them that were only one word (e.g. `Home` -> `BaseHome`) because it felt weird otherwise to just import it as a single uppercase later.

I will wait for some more input and based on that we'll see which option is preferred by everyone

**Comment by xmo-odoo** [comment]
Why do you need to rename the classes? Does the lint trigger because "class in scope" -> function redefined when it's inherited? Seems saddumb...

Also really not fond of renaming them, I'd rather we either:

* imported one symbol up e.g. 

    ```python
    from odoo.addons.account.controllers.portal import PortalAccount
    class PortalAccount(PortalAccount):
    ```
    ->
    ```python
    from odoo.addons.account.controllers import portal
    class PortalAccount(portal.PortalAccount):
    ```

* or flagged existing such case as pylint: disable

File: addons/sale_management/controllers/portal.py [hunk 3]
```diff
@@ -7,10 +7,10 @@
 from odoo.tools import formatLang
 from odoo.exceptions import AccessError, MissingError
 from odoo.http import request
-from odoo.addons.sale.controllers.portal import CustomerPortal
+from odoo.addons.sale.controllers.portal import CustomerPortal as CP
```
Comment (xmo-odoo) [inline addons/sale_management/controllers/portal.py:10]: fbi

Comment (Elkasitu) [reply addons/sale_management/controllers/portal.py:10]: stands for Cheese Pizza of course...

**Comment by Elkasitu** [comment]
> Why do you need to rename the classes? Does the lint trigger because "class in scope" -> function redefined when it's inherited? Seems saddumb...

Indeed, feelsbadman, but to be fair there were one or two cases where it was indeed confusing (see website's `ir_http.py` and `_get_converters` method), however in most cases it doesn't really matter

To sum up the options are:

1. Renaming the inheriting classes: this implies also changing `super()` calls and every single downstream usage of the renamed class
2. Giving them an alias (potentially a more meaningful one like @tde-banana-odoo suggested)
3. Importing the python module that implements the class and using that as a namespace (`foo.MyClass` instead of `MyClass`)
4. Using pylint-disable

I really like #3 the most as you suggested, it's a nice balance of readability and amount of effort needed to implement for existing code, using `pylint-disable` might encite developers in the future to simply copy the same pylint directive without knowing why their code is failing whereas I would like them to follow option #1 or #3 as "conventions" for new developments

File: addons/sale_management/controllers/portal.py (hunk 3 above)
Comment (xmo-odoo) [reply addons/sale_management/controllers/portal.py:10]: pizza

**Comment by xmo-odoo** [comment]
I'm fine with #3 or #2 as long as the names are actually meaningful e.g. not sure that calling everything BasePortal qualifies.

File: addons/sale_management/controllers/portal.py (hunk 3 above)
Comment (Elkasitu) [reply addons/sale_management/controllers/portal.py:10]: didn't find cheese pizza but did find mayan "calander" :thinking:

Comment (xmo-odoo) [reply addons/sale_management/controllers/portal.py:10]: typo, they meant colander
mayan_colander

**Review by rco-odoo — APPROVED** [review]
Good job!

**Comment by rco-odoo** [comment]
@robodoo rebase-ff

**Comment by rco-odoo** [comment]
@robodoo delegate=Elkasitu

**Comment by nhomar** [comment]
@moylop260

**Comment by Elkasitu** [comment]
robodoo r+
